rpc/gnmi/mixed-schema.md

```diff
@@ -108,50 +109,187 @@ if all transactions succeed should the `SetResponse` indicate success. If any
 of the transactions fail, the contents of all origins MUST be rolled back, and
 an error status returned upon responding to the `Set` RPC.
 
-## Example
+## Special considerations for a setRequest with openconfig and cli origins
+
+### Use cases
+The operational use cases for a setRequest `replace` operation containing both OC and CLI origins include:
+
+1. The ability to support incremental transition from CLI based configuration to OpenConfig configuration.
+2. A device may implement configuration items which are only defined in CLI while other configuration items are supported with OpenConfig.
+3. The desire to use a full configuration replacement strategy.
+
+To support these use cases a special meaning to a setRequest with replace origin `cli` and replace origin `openconfig` is defined.
+
+### SetRequest behavior
+
+A `SetRequest` message using the `replace` operation for both `cli` and `openconfig` origins SHOULD be performed as follows:
 
-### Example: OpenConfig and CLI Data
+1. `origin` `cli` SHOULD appear first.  
+2. The entire contents of the `ascii` encoded `cli` data SHOULD appear in a single update.
+3. The `replace` operation for origin `cli` SHOULD be interpeted as a full configuration replacement.  In other words, the target device SHOULD erase the `cli` configuration and replace it with the contents of the `cli` update message.
+4. `origin` `openconfig` should appear second.  
+5. The target device should merge the `cli` and `openconfig` and apply the merged configuration as a single replace operation, replacing the combined `cli` and `openconfig` configuration of the device.
```

Comment:
Sorry in advance - what is the expected behavior in the event of an overlap between them? :)

Advance to step 6?

Reply:
Yes step 6, the gNMI server SHOULD respond with an error.   Note in step 6 the reference to transaction behavior.  If the configuration cannot be fully applied, the entire configuration change MUST be rolled back / not applied.

Reply:
In some previous discussions we talked about two potential behaviors when there is overlap:
1) native model data takes precedence (regardless of the order of native origin & OC origin in set request(s))
or
2) error 
Should we consider allowing either of those behaviors?

Reply:
There are also 2 (maybe more) cases to consider for overlap:
1) The OC and the Native leaf are set to different values
2) The OC and the Native leaf are both explicitly set, but with the same value. 
(it may be simpler to have the same behavior for both of those situations?).
We may also need to define behavior around "default" values (i.e. if Native has a default for a leaf, and the set contains an explicit value for the equivalent leaf, is that "overlap" ?).

Reply:
@dplore I wonder if by using SHOULD term here we allow NOSes that treat Openconfig schema as a translation layer on top of native to **not** error?  

In SR Linux, the merge between CLI and Openconfig can be done for overlapping fields -> data that is applied on top will rewrite the existing data. In the case of the ordering used in this case - (1st)CLI 2(nd)OC - data provided with OC schema will overwrite the overlapping data that was initially pushed with CLI.

As I understand, this whole case of overlapping schemas is a guarding rail, as operationally such a combination in a single set request doesn't make much sense.